run/opencl/opencl_misc.h

```diff
@@ -591,75 +596,42 @@ inline int memmem_pc(const void *haystack, size_t haystack_len,
 #define STRINGIZE(s) STRINGIZE2(s)
 
 /*
- * The reason the functions below are macros is it's the only way we can use
- * them regardless of memory type (eg. __local or __global). The downside is
- * we can't cast them so we need eg. dump8_le for a char array, or output will
- * not be correct.
+ * The below macros need to be called with pointer already cast to
+ * uchar pointer of its type - such as "(__constant uchar*)salt->u"
+ * because type is unknown to us so void* can't help us.
  */
 
-/* Dump an array (or variable) as hex */
-#define dump(x)   dump_stuff_msg(STRINGIZE(x), x, sizeof(x))
-#define dump_stuff(x, size) dump_stuff_msg(STRINGIZE(x), x, size)
+#define dump_le(x, size) dump_stuff_msg(STRINGIZE(x), x, size)
+#define dump_be(x, size) dump_stuff_be_msg(STRINGIZE(x), x, size)
+#define dump_be64(x, size) dump_stuff_be64_msg(STRINGIZE(x), x, size)
 
-/*
- * This clumsy beast finally hides the problem from user.
- */
 #define dump_stuff_msg(msg, x, size) do {	  \
-		switch (sizeof((x)[0])) { \
-		case 8: \
-			dump_stuff64_msg(msg, x, size); \
-			break; \
-		case 4: \
-			dump_stuff32_msg(msg, x, size); \
-			break; \
-		case 2: \
-			dump_stuff16_msg(msg, x, size); \
-			break; \
-		case 1: \
-			dump_stuff8_msg(msg, x, size); \
-			break; \
-		} \
-	} while (0)
-
-/* requires char/uchar */
-#define dump_stuff8_msg(msg, x, size) do {	  \
-		uint ii; \
 		printf("%s : ", msg); \
-		for (ii = 0; ii < (uint)size; ii++) { \
-			printf("%02x", (x)[ii]); \
-			if (ii % 4 == 3) \
+		for (uint xedni_ = 0; xedni_ < (uint)size; xedni_++) { \
+			printf("%02x", (x)[xedni_]); \
+			if (xedni_ % 4 == 3) \
```

Comment:
Why the name `xedni_`?

Reply:
I assume I wouldn't be protected from someone calling with eg. `buffer[ii]` as `x` ending up as `printf("%02x", (buffer[ii])[ii]);`. Or would I?

Reply:
You're right, you need a weird name here also for that reason.